### PR TITLE
feat: introduce `eslint-plugin`, correct eslint config in repo

### PR DESCRIPTION
**Pull request by DarkPurple141**
Hi @danieldelcore - PR could potentially be split into maintenance fixes first and then add `eslint-plugin` later.

** Note a lot of the file changes are whitespace changes for things that weren't previously linted.

Cheers

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,31 +1,44 @@
 module.exports = {
```
Comment (DarkPurple141) [inline .eslintrc.js:1]: Whitespace changes here are related to prettier actually working on the file.

File: .eslintrc.js [hunk 2]
```diff
@@ -1,31 +1,44 @@
...
+    // TODO this is dangerous to allow - should re-address
+    '@typescript-eslint/ban-ts-comment': 'off',
+    '@typescript-eslint/no-unused-vars': [
+      'warn',
+      {
+        ignoreRestSiblings: true,
+      },
+    ],
+  },
+  overrides: [
```
Comment (DarkPurple141) [inline .eslintrc.js:34]: specific config changes related to a number of valid violations in tests

File: .prettierrc.js [hunk 3]
```diff
@@ -4,4 +4,5 @@ module.exports = {
   singleQuote: true,
   printWidth: 80,
   tabWidth: 2,
+  arrowParens: 'avoid',
```
Comment (DarkPurple141) [inline .prettierrc.js:7]: This has been added as the repo appeared to mostly use this convention.

File: package.json [hunk 4]
```diff
@@ -75,5 +75,8 @@
     "packages": [
       "packages/*"
     ]
+  },
+  "engines": {
+    "node": ">=14"
```
Comment (DarkPurple141) [inline package.json:80]: actions run on 14, may as well specify it, also means local dev respects dev deps.

File: package.json [hunk 5]
```diff
@@ -42,27 +42,27 @@
     "@types/jest": "^26.0.15",
     "@types/jscodeshift": "^0.11.0",
     "@types/tar": "^4.0.4",
-    "@typescript-eslint/eslint-plugin": "^1.4.2",
-    "@typescript-eslint/parser": "^1.4.2",
+    "@typescript-eslint/eslint-plugin": "^5.0.0",
```
Comment (DarkPurple141) [inline package.json:45]: these were way behind

File: packages/eslint-plugin/src/__tests__/no-codemod-comment.test.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
+import { RuleTester } from 'eslint';
```
Comment (DarkPurple141) [inline packages/eslint-plugin/src/__tests__/no-codemod-comment.test.ts:1]: Example rule targeting codemod comments

File: packages/eslint-plugin/src/hash.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
+/**
```
Comment (DarkPurple141) [inline packages/eslint-plugin/src/hash.ts:1]: This should be its own package so it can be used in both codemod comment generation and linting

**Comment by danieldelcore** [comment]
Hey @DarkPurple141 sweet cheers for this. I can't see the source for the eslint plugin? Is it coming in a separate PR?

**Review by danieldelcore — APPROVED** [review]

**Comment by DarkPurple141** [comment]
Yeah man I removed from this PR for simplicity

**Pull request merged**